.github/workflows/ci-sdk.yaml

```diff
@@ -61,13 +61,14 @@ jobs:
         test:
         - name: Unit Tests
           run: cargo test $(cat service-with-tests)
-          working-directory: aws-sdk
         - name: Docs
           run: cargo doc --no-deps --document-private-items
-          working-directory: aws-sdk
         - name: Clippy
           run: cargo clippy
-          working-directory: aws-sdk
+        - name: Unused Dependencies
```

Comment:
using the matrix here is genius but also it takes me a couple of minutes of confusion to re-comprend each time I come back to it (especially because it looks like regular `steps`)

Can you add a comment at the top of the matrix explaining how it functions?